### PR TITLE
Fix computer accessories sync issue

### PR DESCRIPTION
**Pull request by abhinav098**
### What does this PR do?

BUG: Fixes sync issue for computer_accessory, and issues found while testing. 

### Impacted Areas

Item Creation

### CODE REVIEW

File: app/models/concerns/subform_utilities.rb [hunk 1]
```diff
@@ -6,8 +6,13 @@ module SubformUtilities
   end
 
   module ClassMethods
-    def distinct_by_column(name)
-      select("distinct on (#{self.name.underscore.pluralize}.#{name}) #{self.name.underscore.pluralize}.*")
+    def distinct_by_column(column)
+      raise StandardError.new("Column name invalid!") unless column_names.include?(column)
+      tbl_name = name.tableize
+      sql_select = send(
+        :sanitize_sql, ["DISTINCT ON (#{tbl_name}.#{column}) #{tbl_name}.*"]
```
Comment (abulsayyad123) [inline app/models/concerns/subform_utilities.rb]: Not sure what `sanitize_sql` does ?

Comment (abhinav098) [reply app/models/concerns/subform_utilities.rb]: It creates a sql query. For safety it's done @abulsayyad123 .

Comment (abhinav098) [reply app/models/concerns/subform_utilities.rb]: https://apidock.com/rails/ActiveRecord/Sanitization/ClassMethods/sanitize_sql_for_conditions

`sanitize_sql` is an alias for `sanitize_sql_for_conditions`

**Comment by namrataukirde** [comment]
:+1:

**Comment by steveyken** [comment]
@abulsayyad123 I mentioned these links to @abhinav098 but please make sure you read and understand them too as it covers the topic of SQL injection - why we need to use `sanitize_sql` methods and write the SQL carefully when adding in `params` input

https://guides.rubyonrails.org/security.html#sql-injection
https://rails-sqli.org/

Little Bobby Tables

**Pull request merged**